### DISCUSSION TITLE
translator proposition

### DISCUSSION THREAD

**Discussion by MontyQI** [Eloston/ungoogled-chromium#1892] [Ideas]
Maybe you can add translator using lingva ([https://github.com/TheDavidDelta/lingva-translate](url)). It's front end to google translator.

**Comment by johannesrld** [comment]
if this browser would support translation i think it would be better to support [bergamont](https://browser.mt/) so you dont send data to servers at all

**Comment by bkil** [comment]
At a minimum, there should be a setting to choose between online translation services. There exist quite a few alternatives for many language pairs:

* https://en.wikipedia.org/wiki/Comparison_of_machine_translation_applications

(Missing from this list: LibreTranslate.com TranslateCat.com Translate.com online-translator.com reverso.net and a bunch of dictionary providers)

There already exist a few frameworks on which you may build offline translator applications for your own language pairs. However, the quality of all of these home-brew translators will be appalling, unless you are also willing to fund someone full time who is fluent in each language pair for fine tuning and ongoing maintenance.

Although, I think still think that offering a minimal viable offline alternative having a lesser quality would still be better than nothing.
